### DISCUSSION TITLE
英文环境下章节标题脚注不能正确编译

### DISCUSSION THREAD

**Discussion by tjhequn** [tuna/thuthesis#906] [Q&A] [answered]
版主大大，小白求问。我准备用这个模板写我的英文论文，所以在使用前对模板做了简短的测试，发现了标题脚注在英文环境下不能正确编译的问题。具体如下：

我直接git clone 这个模板。然后添加了 language=english。如下：
![image](https://github.com/tuna/thuthesis/assets/28583497/d74287c4-9aae-40b3-a58d-3a8b1eb9c94b)

然后我在第一章的标题上添加了脚注的命令， 如下：
![image](https://github.com/tuna/thuthesis/assets/28583497/b7560db0-b912-4737-b10b-72e3885d5025)

然后就编译报错，具体错误如下：

Chapter 1
./data/chap01.tex:3: Use of \@xfootnote doesn't match its definition.
<argument> \@xfootnote \s__text_recursion_tail 

但是把语言改成默认的中文就可以编译出正确的结果，中文的如下：
![image](https://github.com/tuna/thuthesis/assets/28583497/3d1d0fc3-3bf9-4207-9989-6f6028715882)

小白求问，这个问题有什么好的办法可以解决吗？谢谢。

**Comment by zepinglee** [comment, marked as answer]
我能复现你的问题，这个需要深入底层分析。

目前没有找到别的解决办法。

**Reply by tjhequn** [reply]
谢谢大佬，深夜来回复我的问题。

**Reply by zepinglee** [reply]
这个本质上是 `\MakeUppercase` 的展开控制问题，与本模板无关。找到了两种方法：

```tex
\documentclass[degree=doctor,language=english]{thuthesis}
\begin{document}
\chapter[Foo]{Foo\NoCaseChange{\footnote{Bar}}}
\chapter[Foo]{Foo\protect\footnotemark} \footnotetext{Bar}
\end{document}
```

**Reply by tjhequn** [reply]
大佬牛的，测试了一下，可以完美解决这个问题。谢谢大佬
